core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java

```diff
@@ -154,4 +156,48 @@ public static boolean isDSWDatasource( Domain domain ) {
     }
     return !isMetadataDatasource( domain );
   }
+
+  protected String isMetaDataSource( String id ) {
+    Domain domain;
+    try {
+      domain = metadataDomainRepository.getDomain( id );
+      if ( domain == null ) {
+        return null;
+      }
+    } catch ( Exception e ) { // If we can't load the domain then we MUST return false
+      return null;
+    }
+    return isMetadataDatasource( domain ) ? id : null;
+  }
+
+  protected int noOfThreads() {
+    IPluginResourceLoader resLoader = PentahoSystem.get( IPluginResourceLoader.class, null );
+    String noOfThreadsAsString = null;
+    int noOfThreads = 1;
+    try {
+      noOfThreadsAsString =
+              resLoader.getPluginSetting( getClass(), "settings/threads-for-user-session-cache" ); //$NON-NLS-1$
+    } catch ( Exception e ) {
+      e.printStackTrace();
+    }
+    if ( StringUtils.isNotBlank( noOfThreadsAsString ) ) {
+      noOfThreads = Integer.parseInt( noOfThreadsAsString );
+    } else {
+      noOfThreads = Runtime.getRuntime().availableProcessors();
+    }
+    return noOfThreads;
+  }
+
+  protected String isDataSourceWizard( String id ) {
```

Comment:
normally when functions start with "is" they answer a question and the return value is boolean. just in like in `DatasourceService#isMetadataDatasource(domain)`

This function is return a string, the id of the DSW, so the method signature should get something like
`getDataSourceWizardId(id)` or 'getIdDataSourceWizard`